apps/studio/data/api-keys/api-key-delete-mutation.ts

```diff
@@ -0,0 +1,59 @@
+import { useMutation, UseMutationOptions, useQueryClient } from '@tanstack/react-query'
+import { del, handleError } from 'data/fetchers'
+import { toast } from 'sonner'
+import type { ResponseError } from 'types'
+import { apiKeysKeys } from './keys'
+
+export type APIKeyDeleteVariables = {
+  projectRef?: string
+  id: string
+}
+
+export async function deleteAPIKey(payload: APIKeyDeleteVariables) {
+  if (!payload.projectRef) throw new Error('projectRef is required')
+
+  const { data, error } = await del('/v1/projects/{ref}/api-keys/{id}', {
+    params: {
+      path: { ref: payload.projectRef, id: payload.id },
+      // remove this query parameter
```

Comment:
just to check in on this one - still need?

Reply:
it doesn't matter, i'll remove the comment

Reply:
changed to `false` as well